src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c

```diff
@@ -38,14 +39,142 @@ static InitChainEntry D_80ABD750[] = {
+    } else {
+        CutsceneManager_Queue((s16)this->dyna.actor.csId);
+    }
+}
+
+extern f32 D_80ABD778[];
+void ObjTokeiTobira_Update(Actor* thisx, PlayState* play) {
+    s32 pad;
+    ObjTokeiTobira* this = THIS;
+    Player* player = GET_PLAYER(play);
+    s32 pad50;
+    s32 temp_param = this->dyna.actor.params & 1;
+    f32 sp48 = D_80ABD778[temp_param];
+    s32 pad44;
+
+    if (player->actor.bgCheckFlags & 0x200) {
+        if (DynaPoly_GetActor(&play->colCtx, player->actor.wallBgId) == &this->dyna) {
+            f32 sp40;
+            Vec3f sp34;
+
+            Actor_OffsetOfPointInActorCoords(&this->dyna.actor, &sp34, &player->actor.world.pos);
+            sp40 = sp34.x * sp48;
+
+            if (sp40 > 20.0f) {
+                if (sp34.z > 0.0f) {
+                    this->unk160 += (sp40 - 20.0f) * sp48 * 5.3333335f;
+                    this->unk168 = 1;
+                    if (this->unk16C <= 0) {
+                        Actor_PlaySfx(&this->dyna.actor, 0x5809);
+                        this->unk16C = 0x50;
+                    }
+                }
+
+                if (sp40 > 48.0f) {
+                    ObjTokeiTobira* tobira;
+                    if (temp_param == 0) {
+                        tobira = (ObjTokeiTobira*)this->dyna.actor.child;
+                    } else {
+                        tobira = (ObjTokeiTobira*)this->dyna.actor.parent;
+                    }
+
+                    if (tobira != NULL) {
+                        tobira->unk168 = 1;
+                    }
+                }
+            }
+        }
+    } else if (this->unk168 == 1) {
+        Vec3f sp24;
+
+        Actor_OffsetOfPointInActorCoords(&this->dyna.actor, &sp24, &player->actor.world.pos);
+
+        if ((sp24.z > 0.0f) && (sp24.z < 30.0f)) {
+            this->unk168 = 1;
+            this->unk160 += sp48 * 290.0f;
+        }
+    }
+
+    if (this->unk16C > 0) {
+        this->unk16C -= 1;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tokei_Tobira/ObjTokeiTobira_Destroy.s")
+    this->unk168 = 0;
+    this->unk160 *= 0.87f;
+    this->unk164 += this->unk160;
+    if (this->unk164 * sp48 > 14336.0f) {
+        this->unk160 *= 0.1f;
+        this->unk164 = 14336.0f * sp48;
+    } else if (this->unk164 * sp48 < -1000.0f) {
+        this->unk160 *= 0.1f;
+        this->unk164 = -1000.0f * sp48;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tokei_Tobira/func_80ABD3B0.s")
+    this->dyna.actor.shape.rot.y = (s32)this->unk164 + this->dyna.actor.home.rot.y;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tokei_Tobira/ObjTokeiTobira_Update.s")
+    if (this->actionFunc != NULL) {
+        this->actionFunc(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tokei_Tobira/ObjTokeiTobira_Draw.s")
+extern UNK_TYPE D_80ABD780;
+void ObjTokeiTobira_Draw(Actor* thisx, PlayState* play) {
+    Gfx_DrawDListOpa(play, (&D_80ABD780)[thisx->params & 1]);
```

Comment:
`Gfx_DrawDListOpa` takes a `Gfx*`, so  `D_80ABD780` is probably an array of `Gfx*`, which is written like this
```c
Gfx* D_80ABD780[];
```

Reply:
done??